lib/post_creator.rb

```diff
@@ -136,6 +136,14 @@ def valid?
       return false unless skip_validations? || validate_child(topic_creator)
     else
       @topic = Topic.find_by(id: @opts[:topic_id])
+
+      if @topic.present?
+        if @opts[:archetype] == Archetype.private_message
```

Comment:
PSA: We added this test to ensure that any content that were supposed to be "private" never gets public.